cc/tflite_micro/README.md

```diff
@@ -0,0 +1,169 @@
+|   |
+|   |-- debug_log.cc
+|   |   Implement Oak-specific debug logging required by tflm micro-printf.
+|
+|-- tools
+|   |-- BUILD
+|   |
+|   |-- generate_cc_arrays.py
+|   |   Convert a tflite model to a static C array and generate corresponding header and
+|   |   source files which can be specified in cc_binary/cc_library rules. Sourced and
+|   |   modified from tflm's generate_cc_arrays.py.
+|
+|-- BUILD
+|   Provide a build target for generated tflm sources to avoid adding any changes to the
+|   generated folder.
+```
+
+\*
+[Android bionic libc](https://android.googlesource.com/platform/bionic/+/refs/heads/master)\
+\*\* [Google nanolibc](https://github.com/google/nanolibc)
+
+## Build Model Binaries
+
+There are two sorts of freestanding binaries, using hello_world model app as
+example:
+
+1. Build the binary that runs on PC
+
+   ```bash
+   bazel build //cc/tflite_micro/oak/apps/hello_world:hello_world_bin
+   ```
+
+   The binary is built with -nostdlib which removes dependencies of standard
+   libraries i.e. libc, libgcc, etc to ensure a freestanding binary is generated
+   and can run on PC. The binary is good for tflite model porting, debugging and
+   tensor activation validating. Its freestanding nature makes it a good fit of
+   validating execution on TEE.
+
+1. Binaries that runs on Oak server
+   ```bash
+   cd oak_tensorflow_freestanding_bin
+   cargo build
+   ```
+   The binary is built with Oak Restricted Kernel and Oak TensorFlow Service,
+   which can be loaded into virtual machine for execution under TEE.
+
+### Additional Build Options
+
+There are few optional build options provided:
+
+- `--define=no_opt=1`\
+   Disable compiler optimizations when building model app freestanding binaries.
+
+- `--define=no_sse=1`\
+   Disable using streaming SIMD instructions i.e. SSE2/SSSE3/SSE4/AVX2. The
+  option under the hook would use non-optimized nanolibc string apis i.e.
+  str{len|cmp|cpy|...}, mem{cpy|cmp|move|set|...}, etc. Additinally, tflite is
+  built without `-msse4.2`.
+
+- `--define=use_custom_output=1`\
+   To build tflm without using Oak debug_log.cc; instead, using a custom
+  debug_log.cc for multiple use cases i.e. testing, ported to other operating
+  systems that implement proprietary debug logging, etc.
+
+## Debugging Model Binaries
+
+As what aforementioned, we are able to build freestanding libraries that can run
+on PC. In order for debugger i.e. gdb/lldb to correctly work on the binaries, we
+need to specify additional bulid parameters telling bazel to keep all debug
+symbols and optionally disabling optimizations so breakpoints can be precisely
+mapped to correct lines of source code.
+
+Use hell_world model app as example,
+
+```bash
+bazel build --copt=-g --strip=never --define=no_opt=1 //cc/tflite_micro/oak/apps/hello_world:hello_world_bin
+```
+
+`--define=no_opt=1` is optional if correct source mapping is not needed.
+
+Next, configure i.e. lldb for VS Code,\
```

Comment:
Not sure about those `\` here and in other places, can they be removed?

Reply:
An additional empty line will be added without `\`.
I wanted them to show in a following line but I'm also ok to remove them.

Reply:
Done removing them.